src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -240,6 +283,41 @@ private static int GetCurrentProcessId()
             return Interop.Sys.GetPid();
         }
 
+        /// <summary>
+        /// Returns all immediate child processes.
+        /// </summary>
+        private IReadOnlyList<Process> GetChildProcesses()
+        {
+            var childProcesses = new List<Process>();
+
+            foreach (Process possibleChildProcess in GetProcesses())
+            {
+                var keep = false;
+
+                try
+                {
+                    if (IsParentOf(possibleChildProcess))
+                    {
+                        childProcesses.Add(possibleChildProcess);
+                        keep = true;
+                    }
+                }
+                finally
+                {
+                    if (!keep)
+                    {
+                        possibleChildProcess.Dispose();
+                    }
+                }
+            }
+
+            return childProcesses;
+        }
+
+        /// <summary>Checks whether the argument is a direct child of this process.</summary>
+        private bool IsParentOf(Process possibleChildProcess) =>
```

Comment:
PIDs must get recycled on Unix just like Windows. How do you avoid the "start time" issue you deal with on Windows?

Reply:
On Windows, a process's parent ID is set when the process starts. After that point, it's never updated, even if the parent dies. This never-updated behavior necessitates checking start time to ensure we have the true parent and not just a process that has the same ID as the deceased parent (because the parent PID reported by the child might now be assigned to a new process).

Linux, on the other hand, updates a child's parent process ID when its parent dies. As long as we make sure the parent process doesn't die while we're working with it, the parent PID should be pointing to the correct parent process, saving us from needing to do the start time check.